model/model.go

```diff
@@ -0,0 +1,69 @@
+package model
+
+import "gorm.io/gorm"
+
+const (
+	RawTax float64 = 12.5
+
+	ImportTax     float64 = 10
+	Surcharge100  float64 = 5
+	Surcharge200  float64 = 10
+	SurchargeMore float64 = 5
+
+	ManufacturedTax   float64 = 12.5
+	ManufacturedExtra float64 = 2
+)
+
+//every type of item should implement this
+type Item interface {
+	GetDetails() (string, float64, int, float64, float64)
+	Calc() float64
+}
+
+type BaseItem struct {
+	gorm.Model
+	Name     string
+	Price    float64
+	Quantity int
+	Total    float64
+	Typ      string
+}
+
+func (b *BaseItem) GetDetails() (name string, price float64, quantity int, total float64) {
```

Comment:
what is the use of this func, why we cant directly use base item struct everywhere